src/vm/jvm/runtime/org/raku/nqp/sixmodel/reprs/DecoderInstance.java

```diff
@@ -156,7 +159,8 @@ public synchronized String takeAllChars(ThreadContext tc) {
             decoder.flush(target);
             decoder.reset();
         }
-        return Normalizer.normalize(decodedBuffer(target), Normalizer.Form.NFC);
+        String normalized = Normalizer.normalize(decodedBuffer(target), Normalizer.Form.NFC);
+        return translate_newlines ? normalized.replace("\r\n", "\n") : normalized;
```

Comment:
I think this should happen in `takeChars` and `takeAvailableChars` too, probably by factoring out this line into a method. Otherwise it seems to be doing the right thing.

Reply:
Cool, I'll give that a shot and see what happens.

Reply:
Done. No change in tests.